api/app/src/routes/settings.ts

```diff
@@ -78,4 +103,48 @@ router.post(
   })
 );
 
+/** ---------------------------------------------------------------------------
+ * GET /settings/webhook
+ *
+ * Get webhook status information.
+ *
+ * @return {WebhookStatusDTO} The Webhook status information or empty body if no
+ * no settings defined.
+ */
+router.get(
+  "/webhook",
+  asyncHandler(async (req: Request, res: Response) => {
+    const id = getCxIdOrFail(req);
+    const settings = await getSettings({ id });
+    if (!settings) return res.status(status.OK).send({});
+    const { processing, failure } = await countFailedAndProcessingRequests(id);
+    res
+      .status(status.OK)
+      .json(
+        WebhookStatusDTO.fromEntity(
+          settings.webhookEnabled,
+          settings.webhookStatusDetail,
+          processing,
+          failure
+        )
+      );
+  })
+);
+
+/** ---------------------------------------------------------------------------
+ * POST /settings/webhook/retry
+ *
+ * Retries failed webhook requests.
+ *
+ * @return {200} indicating retry being processed.
+ */
+router.post(
+  "/webhook/retry",
+  asyncHandler(async (req: Request, res: Response) => {
+    const cxId = getCxIdOrFail(req);
+    await retryFailedRequests(cxId);
```

Comment:
A couple things to consider:

- What happens if this is called quickly multiple times in succession - is this idempotent?
- Does sequelize automatically handle optimistic locking or something to that effect, or is there a chance that there can be multiple processes running on the same failed webhooks, before one transaction had a chance to commit the rows to `processing` ?

Reply:
When there's a request to `/retry` we first update the WH requests to `processing`, this minimizes problems with subsequent requests reprocessing the same WH request again.

To 100% avoid any concurrency issues we'd need to implement a logical semaphore so only one request could process WHs (for a given customer) at any given time.

This should be solved once we decouple preparing WH requests from actually running them (SQS, exponential backoff, single processing per customer, etc.).